### PR TITLE
Add flags for prometheus-to-sd components.

### PR DESCRIPTION
**Pull request by loburm**
Configure prometheus-to-sd-endpoint and prometheus-to-sd-prefix base on
the environment.

**Release note**:
NONE

### CODE REVIEW

File: cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml
Comment (piosz) [inline cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml]: As discussed offline please remove this.

File: cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml
Comment (piosz) [inline cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml]: just addons

**Review by piosz — CHANGES_REQUESTED** [review]
please apply comments in other places as well

File: cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml
Comment (loburm) [inline cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml]: That may be not a very good idea. It may happen that some people already configured alerts based on this name.

File: cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml
Comment (loburm) [inline cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml]: Removed

**Comment by loburm** [comment]
Added one more env variable that allow user to completely disable prometheus-to-sd.

File: cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml
Comment (loburm) [inline cluster/addons/cluster-monitoring/stackdriver/heapster-controller.yaml]: Sorry, I haven't understood. We can still specify the correct prefix and metric name will not change.

Fixed.

**Review by piosz — APPROVED** [review]

**Comment by piosz** [comment]
/lgtm
The logic does what we would like to achieve though it's a bit hack. Waiting for @mikedanese to review it.

**Comment by loburm** [comment]
I have rebased PR (it was two weeks old). Can I get lgtm once again?

Thanks

**Pull request merged**
